tests/ui/traits/const-traits/const-trait-async-assoc-fn.rs

```diff
@@ -0,0 +1,10 @@
+//@ edition: 2021
+#![feature(const_trait_impl)]
+
+#[const_trait]
+trait Tr {
```

Comment:
Could you use the new const trait syntax (#143879). The attr-based one is legacy and is only kept because we use a beta (or nightly?) rustfmt for the stdlib instead of a locally built one, so it doesn't "yet" understand the new syntax.

```suggestion
const trait Tr {
```